File: src/data/phonemeExampleWords.ts
import type { WordExample } from './phonemes.types';

/**
 * Three example words per recorded IPA symbol, with the highlighted
 * range being the letter(s) that spell that phoneme in that word.
 * Migrated from lib/core/content/phoneme_example_words.dart.
 */
export const phonemeExampleWords: Record<string, WordExample[]> = {
  'iː': [
    { word: 'see', highlightStart: 1, highlightLength: 2, audioSlug: 'see' },
    { word: 'tree', highlightStart: 2, highlightLength: 2, audioSlug: 'tree' },
    { word: 'key', highlightStart: 1, highlightLength: 2, audioSlug: 'key' },
  ],
  'ɪ': [
    { word: 'sit', highlightStart: 1, highlightLength: 1, audioSlug: 'sit' },
    { word: 'fish', highlightStart: 1, highlightLength: 1, audioSlug: 'fish' },
    { word: 'big', highlightStart: 1, highlightLength: 1, audioSlug: 'big' },
  ],
  'ɛ': [
    { word: 'bed', highlightStart: 1, highlightLength: 1, audioSlug: 'bed' },
    { word: 'red', highlightStart: 1, highlightLength: 1, audioSlug: 'red' },
    { word: 'ten', highlightStart: 1, highlightLength: 1, audioSlug: 'ten' },
  ],
  'æ': [
    { word: 'cat', highlightStart: 1, highlightLength: 1, audioSlug: 'cat' },
    { word: 'hat', highlightStart: 1, highlightLength: 1, audioSlug: 'hat' },
    { word: 'apple', highlightStart: 0, highlightLength: 1, audioSlug: 'apple' },
  ],
  'ɑ': [
    { word: 'hot', highlightStart: 1, highlightLength: 1, audioSlug: 'hot' },
    { word: 'sock', highlightStart: 1, highlightLength: 1, audioSlug: 'sock' },
    { word: 'top', highlightStart: 1, highlightLength: 1, audioSlug: 'top' },
  ],
  'ɔː': [
    { word: 'ball', highlightStart: 1, highlightLength: 1, audioSlug: 'ball' },
    { word: 'saw', highlightStart: 1, highlightLength: 2, audioSlug: 'saw' },
    { word: 'tall', highlightStart: 1, highlightLength: 1, audioSlug: 'tall' },
  ],
  'ʊ': [
    { word: 'book', highlightStart: 1, highlightLength: 2, audioSlug: 'book' },
    { word: 'foot', highlightStart: 1, highlightLength: 2, audioSlug: 'foot' },
    { word: 'put', highlightStart: 1, highlightLength: 1, audioSlug: 'put' },
  ],
  'uː': [
    { word: 'blue', highlightStart: 2, highlightLength: 2, audioSlug: 'blue' },
    { word: 'moon', highlightStart: 1, highlightLength: 2, audioSlug: 'moon' },
    { word: 'food', highlightStart: 1, highlightLength: 2, audioSlug: 'food' },
  ],
  'ʌ': [
    { word: 'cup', highlightStart: 1, highlightLength: 1, audioSlug: 'cup' },
    { word: 'sun', highlightStart: 1, highlightLength: 1, audioSlug: 'sun' },
    { word: 'run', highlightStart: 1, highlightLength: 1, audioSlug: 'run' },
  ],
  'ə': [
    { word: 'about', highlightStart: 0, highlightLength: 1, audioSlug: 'about' },
    { word: 'sofa', highlightStart: 3, highlightLength: 1, audioSlug: 'sofa' },
    { word: 'banana', highlightStart: 1, highlightLength: 1, audioSlug: 'banana' },
  ],
  'eɪ': [
    { word: 'day', highlightStart: 1, highlightLength: 2, audioSlug: 'day' },
    { word: 'rain', highlightStart: 1, highlightLength: 2, audioSlug: 'rain' },
    { word: 'eight', highlightStart: 0, highlightLength: 4, audioSlug: 'eight' },
  ],
  'aɪ': [
    { word: 'eye', highlightStart: 0, highlightLength: 3, audioSlug: 'eye' },
    { word: 'pie', highlightStart: 1, highlightLength: 2, audioSlug: 'pie' },
    { word: 'fly', highlightStart: 2, highlightLength: 1, audioSlug: 'fly' },
  ],
  'aʊ': [
    { word: 'house', highlightStart: 1, highlightLength: 2, audioSlug: 'house' },
    { word: 'cow', highlightStart: 1, highlightLength: 2, audioSlug: 'cow' },
    { word: 'mouth', highlightStart: 1, highlightLength: 2, audioSlug: 'mouth' },
  ],
  'ɔɪ': [
    { word: 'boy', highlightStart: 1, highlightLength: 2, audioSlug: 'boy' },
    { word: 'toy', highlightStart: 1, highlightLength: 2, audioSlug: 'toy' },
    { word: 'coin', highlightStart: 1, highlightLength: 2, audioSlug: 'coin' },
  ],
  'oʊ': [
    { word: 'go', highlightStart: 1, highlightLength: 1, audioSlug: 'go' },
    { word: 'boat', highlightStart: 1, highlightLength: 2, audioSlug: 'boat' },
    { word: 'snow', highlightStart: 2, highlightLength: 2, audioSlug: 'snow' },
  ],
  'ɝ': [
    { word: 'bird', highlightStart: 1, highlightLength: 2, audioSlug: 'bird' },
    { word: 'girl', highlightStart: 1, highlightLength: 2, audioSlug: 'girl' },
    { word: 'first', highlightStart: 1, highlightLength: 2, audioSlug: 'first' },
  ],
  'ɚ': [
    { word: 'mother', highlightStart: 4, highlightLength: 2, audioSlug: 'mother' },
    { word: 'teacher', highlightStart: 5, highlightLength: 2, audioSlug: 'teacher' },
    { word: 'dinner', highlightStart: 4, highlightLength: 2, audioSlug: 'dinner' },
  ],
  'ɑr': [
    { word: 'car', highlightStart: 1, highlightLength: 2, audioSlug: 'car' },
    { word: 'star', highlightStart: 2, highlightLength: 2, audioSlug: 'star' },
    { word: 'far', highlightStart: 1, highlightLength: 2, audioSlug: 'far' },
  ],
  'ɛr': [
    { word: 'hair', highlightStart: 1, highlightLength: 3, audioSlug: 'hair' },
    { word: 'bear', highlightStart: 1, highlightLength: 3, audioSlug: 'bear' },
    { word: 'chair', highlightStart: 2, highlightLength: 3, audioSlug: 'chair' },
  ],
  'ɪr': [
    { word: 'ear', highlightStart: 0, highlightLength: 3, audioSlug: 'ear' },
    { word: 'deer', highlightStart: 1, highlightLength: 3, audioSlug: 'deer' },
    { word: 'here', highlightStart: 1, highlightLength: 3, audioSlug: 'here' },
  ],
  'ɔr': [
    { word: 'door', highlightStart: 1, highlightLength: 3, audioSlug: 'door' },
    { word: 'four', highlightStart: 1, highlightLength: 3, audioSlug: 'four' },
    { word: 'corn', highlightStart: 1, highlightLength: 2, audioSlug: 'corn' },
  ],
  'p': [
    { word: 'pen', highlightStart: 0, highlightLength: 1, audioSlug: 'pen' },
    { word: 'cup', highlightStart: 2, highlightLength: 1, audioSlug: 'cup' },
    { word: 'apple', highlightStart: 1, highlightLength: 2, audioSlug: 'apple' },
  ],
  't': [
    { word: 'top', highlightStart: 0, highlightLength: 1, audioSlug: 'top' },
    { word: 'cat', highlightStart: 2, highlightLength: 1, audioSlug: 'cat' },
    { word: 'stop', highlightStart: 1, highlightLength: 1, audioSlug: 'stop' },
  ],
  'k': [
    { word: 'cat', highlightStart: 0, highlightLength: 1, audioSlug: 'cat' },
    { word: 'key', highlightStart: 0, highlightLength: 1, audioSlug: 'key' },
    { word: 'book', highlightStart: 3, highlightLength: 1, audioSlug: 'book' },
  ],
  'tʃ': [
    { word: 'chair', highlightStart: 0, highlightLength: 2, audioSlug: 'chair' },
    { word: 'cheese', highlightStart: 0, highlightLength: 2, audioSlug: 'cheese' },
    { word: 'watch', highlightStart: 2, highlightLength: 3, audioSlug: 'watch' },
  ],
  'f': [
    { word: 'fish', highlightStart: 0, highlightLength: 1, audioSlug: 'fish' },
    { word: 'leaf', highlightStart: 3, highlightLength: 1, audioSlug: 'leaf' },
    { word: 'coffee', highlightStart: 2, highlightLength: 2, audioSlug: 'coffee' },
  ],
  'θ': [
    { word: 'think', highlightStart: 0, highlightLength: 2, audioSlug: 'think' },
    { word: 'teeth', highlightStart: 3, highlightLength: 2, audioSlug: 'teeth' },
    { word: 'bath', highlightStart: 2, highlightLength: 2, audioSlug: 'bath' },
  ],
  's': [
    { word: 'sun', highlightStart: 0, highlightLength: 1, audioSlug: 'sun' },
    { word: 'bus', highlightStart: 2, highlightLength: 1, audioSlug: 'bus' },
    { word: 'snake', highlightStart: 0, highlightLength: 1, audioSlug: 'snake' },
  ],
  'ʃ': [
    { word: 'shoe', highlightStart: 0, highlightLength: 2, audioSlug: 'shoe' },
    { word: 'fish', highlightStart: 2, highlightLength: 2, audioSlug: 'fish' },
    { word: 'wash', highlightStart: 2, highlightLength: 2, audioSlug: 'wash' },
  ],
  'b': [
    { word: 'ball', highlightStart: 0, highlightLength: 1, audioSlug: 'ball' },
    { word: 'baby', highlightStart: 0, highlightLength: 1, audioSlug: 'baby' },
    { word: 'tub', highlightStart: 2, highlightLength: 1, audioSlug: 'tub' },
  ],
  'd': [
    { word: 'dog', highlightStart: 0, highlightLength: 1, audioSlug: 'dog' },
    { word: 'red', highlightStart: 2, highlightLength: 1, audioSlug: 'red' },
    { word: 'bed', highlightStart: 2, highlightLength: 1, audioSlug: 'bed' },
  ],
  'g': [
    { word: 'go', highlightStart: 0, highlightLength: 1, audioSlug: 'go' },
    { word: 'dog', highlightStart: 2, highlightLength: 1, audioSlug: 'dog' },
    { word: 'egg', highlightStart: 1, highlightLength: 2, audioSlug: 'egg' },
  ],
  'dʒ': [
    { word: 'jump', highlightStart: 0, highlightLength: 1, audioSlug: 'jump' },
    { word: 'juice', highlightStart: 0, highlightLength: 1, audioSlug: 'juice' },
    { word: 'orange', highlightStart: 4, highlightLength: 1, audioSlug: 'orange' },
  ],
  'v': [
    { word: 'van', highlightStart: 0, highlightLength: 1, audioSlug: 'van' },
    { word: 'love', highlightStart: 2, highlightLength: 1, audioSlug: 'love' },
    { word: 'seven', highlightStart: 2, highlightLength: 1, audioSlug: 'seven' },
  ],
  'ð': [
    { word: 'this', highlightStart: 0, highlightLength: 2, audioSlug: 'this' },
    { word: 'mother', highlightStart: 2, highlightLength: 2, audioSlug: 'mother' },
    { word: 'bathe', highlightStart: 2, highlightLength: 2, audioSlug: 'bathe' },
  ],
  'z': [
    { word: 'zoo', highlightStart: 0, highlightLength: 1, audioSlug: 'zoo' },
    { word: 'lazy', highlightStart: 2, highlightLength: 1, audioSlug: 'lazy' },
    { word: 'nose', highlightStart: 2, highlightLength: 1, audioSlug: 'nose' },
  ],
  'ʒ': [
    { word: 'vision', highlightStart: 2, highlightLength: 2, audioSlug: 'vision' },
    { word: 'treasure', highlightStart: 4, highlightLength: 1, audioSlug: 'treasure' },
    { word: 'measure', highlightStart: 3, highlightLength: 1, audioSlug: 'measure' },
  ],
  'm': [
    { word: 'moon', highlightStart: 0, highlightLength: 1, audioSlug: 'moon' },
    { word: 'mom', highlightStart: 0, highlightLength: 1, audioSlug: 'mom' },
    { word: 'swim', highlightStart: 3, highlightLength: 1, audioSlug: 'swim' },
  ],
  'n': [
    { word: 'nose', highlightStart: 0, highlightLength: 1, audioSlug: 'nose' },
    { word: 'sun', highlightStart: 2, highlightLength: 1, audioSlug: 'sun' },
    { word: 'banana', highlightStart: 2, highlightLength: 1, audioSlug: 'banana' },
  ],
  'ŋ': [
    { word: 'sing', highlightStart: 2, highlightLength: 2, audioSlug: 'sing' },
    { word: 'ring', highlightStart: 2, highlightLength: 2, audioSlug: 'ring' },
    { word: 'king', highlightStart: 2, highlightLength: 2, audioSlug: 'king' },
  ],
  'l': [
    { word: 'lion', highlightStart: 0, highlightLength: 1, audioSlug: 'lion' },
    { word: 'ball', highlightStart: 2, highlightLength: 2, audioSlug: 'ball' },
    { word: 'apple', highlightStart: 3, highlightLength: 1, audioSlug: 'apple' },
  ],
  'w': [
    { word: 'water', highlightStart: 0, highlightLength: 1, audioSlug: 'water' },
    { word: 'wet', highlightStart: 0, highlightLength: 1, audioSlug: 'wet' },
    { word: 'away', highlightStart: 1, highlightLength: 1, audioSlug: 'away' },
  ],
  'j': [
    { word: 'yes', highlightStart: 0, highlightLength: 1, audioSlug: 'yes' },
    { word: 'yellow', highlightStart: 0, highlightLength: 1, audioSlug: 'yellow' },
    { word: 'yard', highlightStart: 0, highlightLength: 1, audioSlug: 'yard' },
  ],
  'h': [
    { word: 'hat', highlightStart: 0, highlightLength: 1, audioSlug: 'hat' },
    { word: 'house', highlightStart: 0, highlightLength: 1, audioSlug: 'house' },
    { word: 'hello', highlightStart: 0, highlightLength: 1, audioSlug: 'hello' },
  ],
  'r': [
    { word: 'run', highlightStart: 0, highlightLength: 1, audioSlug: 'run' },
    { word: 'zero', highlightStart: 2, highlightLength: 1, audioSlug: 'zero' },
    { word: 'rabbit', highlightStart: 0, highlightLength: 1, audioSlug: 'rabbit' },
  ],
  'ʔ': [
    { word: 'uh-oh', highlightStart: 2, highlightLength: 1, audioSlug: 'uh_oh' },
    { word: 'button', highlightStart: 2, highlightLength: 2, audioSlug: 'button' },
    { word: 'kitten', highlightStart: 2, highlightLength: 2, audioSlug: 'kitten' },
  ],
  'ɾ': [
    { word: 'butter', highlightStart: 2, highlightLength: 2, audioSlug: 'butter' },
    { word: 'water', highlightStart: 2, highlightLength: 1, audioSlug: 'water' },
    { word: 'ladder', highlightStart: 2, highlightLength: 2, audioSlug: 'ladder' },
  ],
};

File: lib/core/content/phoneme_example_words.dart
/// One example word for a phoneme, with the substring that spells the
/// sound (so the UI can highlight it) and its own recorded pronunciation.
class WordExample {
  const WordExample(this.word, this.highlightStart, this.highlightLength);

  final String word;
  final int highlightStart;
  final int highlightLength;

  /// Filename slug for assets/audio/example_words/{slug}.m4a.
  String get audioSlug => word.replaceAll('-', '_');
}

/// Three example words per recorded IPA symbol, shown alongside the
/// phoneme tiles on PhoneticsChartPage. Each word's highlighted range is
/// the specific letter(s) that spell that phoneme in that word.
const Map<String, List<WordExample>> phonemeExampleWords =
    <String, List<WordExample>>{
      // Vowels
      'iː': <WordExample>[
        WordExample('see', 1, 2),
        WordExample('tree', 2, 2),
        WordExample('key', 1, 2),
      ],
      'ɪ': <WordExample>[
        WordExample('sit', 1, 1),
        WordExample('fish', 1, 1),
        WordExample('big', 1, 1),
      ],
      'ɛ': <WordExample>[
        WordExample('bed', 1, 1),
        WordExample('red', 1, 1),
        WordExample('ten', 1, 1),
      ],
      'æ': <WordExample>[
        WordExample('cat', 1, 1),
        WordExample('hat', 1, 1),
        WordExample('apple', 0, 1),
      ],
      'ɑ': <WordExample>[
        WordExample('hot', 1, 1),
        WordExample('sock', 1, 1),
        WordExample('top', 1, 1),
      ],
      'ɔː': <WordExample>[
        WordExample('ball', 1, 1),
        WordExample('saw', 1, 2),
        WordExample('tall', 1, 1),
      ],
      'ʊ': <WordExample>[
        WordExample('book', 1, 2),
        WordExample('foot', 1, 2),
        WordExample('put', 1, 1),
      ],
      'uː': <WordExample>[
        WordExample('blue', 2, 2),
        WordExample('moon', 1, 2),
        WordExample('food', 1, 2),
      ],
      'ʌ': <WordExample>[
        WordExample('cup', 1, 1),
        WordExample('sun', 1, 1),
        WordExample('run', 1, 1),
      ],
      'ə': <WordExample>[
        WordExample('about', 0, 1),
        WordExample('sofa', 3, 1),
        WordExample('banana', 1, 1),
      ],

      // Diphthongs
      'eɪ': <WordExample>[
        WordExample('day', 1, 2),
        WordExample('rain', 1, 2),
        WordExample('eight', 0, 4),
      ],
      'aɪ': <WordExample>[
        WordExample('eye', 0, 3),
        WordExample('pie', 1, 2),
        WordExample('fly', 2, 1),
      ],
      'aʊ': <WordExample>[
        WordExample('house', 1, 2),
        WordExample('cow', 1, 2),
        WordExample('mouth', 1, 2),
      ],
      'ɔɪ': <WordExample>[
        WordExample('boy', 1, 2),
        WordExample('toy', 1, 2),
        WordExample('coin', 1, 2),
      ],
      'oʊ': <WordExample>[
        WordExample('go', 1, 1),
        WordExample('boat', 1, 2),
        WordExample('snow', 2, 2),
      ],

      // R-colored vowels
      'ɝ': <WordExample>[
        WordExample('bird', 1, 2),
        WordExample('girl', 1, 2),
        WordExample('first', 1, 2),
      ],
      'ɚ': <WordExample>[
        WordExample('mother', 4, 2),
        WordExample('teacher', 5, 2),
        WordExample('dinner', 4, 2),
      ],
      'ɑr': <WordExample>[
        WordExample('car', 1, 2),
        WordExample('star', 2, 2),
        WordExample('far', 1, 2),
      ],
      'ɛr': <WordExample>[
        WordExample('hair', 1, 3),
        WordExample('bear', 1, 3),
        WordExample('chair', 2, 3),
      ],
      'ɪr': <WordExample>[
        WordExample('ear', 0, 3),
        WordExample('deer', 1, 3),
        WordExample('here', 1, 3),
      ],
      'ɔr': <WordExample>[
        WordExample('door', 1, 3),
        WordExample('four', 1, 3),
        WordExample('corn', 1, 2),
      ],

      // Voiceless consonants
      'p': <WordExample>[
        WordExample('pen', 0, 1),
        WordExample('cup', 2, 1),
        WordExample('apple', 1, 2),
      ],
      't': <WordExample>[
        WordExample('top', 0, 1),
        WordExample('cat', 2, 1),
        WordExample('stop', 1, 1),
      ],
      'k': <WordExample>[
        WordExample('cat', 0, 1),
        WordExample('key', 0, 1),
        WordExample('book', 3, 1),
      ],
      'tʃ': <WordExample>[
        WordExample('chair', 0, 2),
        WordExample('cheese', 0, 2),
        WordExample('watch', 2, 3),
      ],
      'f': <WordExample>[
        WordExample('fish', 0, 1),
        WordExample('leaf', 3, 1),
        WordExample('coffee', 2, 2),
      ],
      'θ': <WordExample>[
        WordExample('think', 0, 2),
        WordExample('teeth', 3, 2),
        WordExample('bath', 2, 2),
      ],
      's': <WordExample>[
        WordExample('sun', 0, 1),
        WordExample('bus', 2, 1),
        WordExample('snake', 0, 1),
      ],
      'ʃ': <WordExample>[
        WordExample('shoe', 0, 2),
        WordExample('fish', 2, 2),
        WordExample('wash', 2, 2),
      ],

      // Voiced consonants
      'b': <WordExample>[
        WordExample('ball', 0, 1),
        WordExample('baby', 0, 1),
        WordExample('tub', 2, 1),
      ],
      'd': <WordExample>[
        WordExample('dog', 0, 1),
        WordExample('red', 2, 1),
        WordExample('bed', 2, 1),
      ],
      'g': <WordExample>[
        WordExample('go', 0, 1),
        WordExample('dog', 2, 1),
        WordExample('egg', 1, 2),
      ],
      'dʒ': <WordExample>[
        WordExample('jump', 0, 1),
        WordExample('juice', 0, 1),
        WordExample('orange', 4, 1),
      ],
      'v': <WordExample>[
        WordExample('van', 0, 1),
        WordExample('love', 2, 1),
        WordExample('seven', 2, 1),
      ],
      'ð': <WordExample>[
        WordExample('this', 0, 2),
        WordExample('mother', 2, 2),
        WordExample('bathe', 2, 2),
      ],
      'z': <WordExample>[
        WordExample('zoo', 0, 1),
        WordExample('lazy', 2, 1),
        WordExample('nose', 2, 1),
      ],
      'ʒ': <WordExample>[
        WordExample('vision', 2, 2),
        WordExample('treasure', 4, 1),
        WordExample('measure', 3, 1),
      ],

      // Other consonants
      'm': <WordExample>[
        WordExample('moon', 0, 1),
        WordExample('mom', 0, 1),
        WordExample('swim', 3, 1),
      ],
      'n': <WordExample>[
        WordExample('nose', 0, 1),
        WordExample('sun', 2, 1),
        WordExample('banana', 2, 1),
      ],
      'ŋ': <WordExample>[
        WordExample('sing', 2, 2),
        WordExample('ring', 2, 2),
        WordExample('king', 2, 2),
      ],
      'l': <WordExample>[
        WordExample('lion', 0, 1),
        WordExample('ball', 2, 2),
        WordExample('apple', 3, 1),
      ],
      'w': <WordExample>[
        WordExample('water', 0, 1),
        WordExample('wet', 0, 1),
        WordExample('away', 1, 1),
      ],
      'j': <WordExample>[
        WordExample('yes', 0, 1),
        WordExample('yellow', 0, 1),
        WordExample('yard', 0, 1),
      ],
      'h': <WordExample>[
        WordExample('hat', 0, 1),
        WordExample('house', 0, 1),
        WordExample('hello', 0, 1),
      ],
      'r': <WordExample>[
        WordExample('run', 0, 1),
        WordExample('zero', 2, 1),
        WordExample('rabbit', 0, 1),
      ],
      'ʔ': <WordExample>[
        WordExample('uh-oh', 2, 1),
        WordExample('button', 2, 2),
        WordExample('kitten', 2, 2),
      ],
      'ɾ': <WordExample>[
        WordExample('butter', 2, 2),
        WordExample('water', 2, 1),
        WordExample('ladder', 2, 2),
      ],
    };
